src/core/hle/service/lm/lm.cpp

```diff
@@ -30,54 +79,257 @@ class ILogger final : public ServiceFramework<ILogger> {
+        ASSERT(length == sizeof(u32));
+        u32_le output{};
+        std::memcpy(&output, data.data() + offset, sizeof(u32));
+        offset += length;
+        return output;
+    }
+
+    u64_le ReadAsU64(const std::vector<u8>& data, std::size_t& offset, std::size_t length) {
+        ASSERT(length == sizeof(u64));
+        u64_le output{};
+        std::memcpy(&output, data.data() + offset, sizeof(u64));
+        offset += length;
+        return output;
+    }
+
+    void ParseLog(const LogPacketHeaderEntrty entry, const std::vector<u8>& log_data) {
+        // Possible entries
+        std::optional<std::string> text_log;
+        std::optional<u32> line_number;
+        std::optional<std::string> file_name;
+        std::optional<std::string> function_name;
+        std::optional<std::string> module_name;
+        std::optional<std::string> thread_name;
+        std::optional<u64> log_pack_drop_count;
+        std::optional<s64> user_system_clock;
+        std::optional<std::string> process_name;
+
+        std::size_t offset{};
+        while (offset < log_data.size()) {
+            const auto key = static_cast<LogDataChunkKey>(ReadLeb128(log_data, offset));
+            const auto chunk_size = ReadLeb128(log_data, offset);
+
+            switch (key) {
+            case LogDataChunkKey::LogSessionBegin:
+            case LogDataChunkKey::LogSessionEnd:
+                break;
+            case LogDataChunkKey::TextLog:
+                text_log = std::move(ReadString(log_data, offset, chunk_size));
+                break;
+            case LogDataChunkKey::LineNumber:
+                line_number = ReadAsU32(log_data, offset, chunk_size);
+                break;
+            case LogDataChunkKey::FileName:
+                file_name = std::move(ReadString(log_data, offset, chunk_size));
+                break;
+            case LogDataChunkKey::FunctionName:
+                function_name = std::move(ReadString(log_data, offset, chunk_size));
+                break;
+            case LogDataChunkKey::ModuleName:
+                module_name = std::move(ReadString(log_data, offset, chunk_size));
+                break;
+            case LogDataChunkKey::ThreadName:
+                thread_name = std::move(ReadString(log_data, offset, chunk_size));
+                break;
+            case LogDataChunkKey::LogPacketDropCount:
+                log_pack_drop_count = ReadAsU64(log_data, offset, chunk_size);
+                break;
+            case LogDataChunkKey::UserSystemClock:
+                user_system_clock = ReadAsU64(log_data, offset, chunk_size);
+                break;
+            case LogDataChunkKey::ProcessName:
+                process_name = std::move(ReadString(log_data, offset, chunk_size));
+                break;
+            }
+        }
+
+        std::string output_log{};
+        if (process_name && !process_name.value().empty()) {
+            output_log += fmt::format("Process: {}\n", process_name.value());
```

Comment:
```suggestion
            output_log += fmt::format("Process: {}\n", *process_name));
```

avoids a validity check, since we know its valid at this point. Ditto for the `valid().empty()` checks as well here and below.